### PR TITLE
cleans up mines

### PR DESCRIPTION
**Pull request by Delimusca**
nominating mines.dm for worst file in the codebase.
only 80 lines and yet 6 pointless procs and two colon operators. and they were dense. completely baffling.
removed radmine and kickmine, i doubt anyone will miss them.
eventually i'd like to make these things disarmable and perhaps craftable like chemnades.

### CODE REVIEW

**Comment by Menshin** [comment]
Would probably be better to overload the _triggermine()_ proc for mines subtypes instead of using a _switch(triggertype)_.

Also i suppose _icon_state = "uglymine"_ for subtypes is a placeholder for potential different mine icons ?

Other than that :+1:

Oh, and try _mecha_ or _washing mashine_ if you want some really bad code.

**Comment by Delimusca** [comment]
i figure a switch is fine for the simple mines since they only do one thing and apparently overrides count towards the proclimit or something.

**Comment by Niknakflak** [comment]
Dont remove kick mines.  Badmins love them to mess with other admins.  They are so unique that no other item works like it.  I love them ;_;

**Comment by Delimusca** [comment]
deleting client with a colon check is very not kosher, paisano.
if you really, REALLY think these things should exist, i can probably put them back in a cleaner fashion.

**Comment by RemieRichards** [comment]
@Delimusca you can put them back without the :.

**Comment by Delimusca** [comment]
i had the terrible idea of making them do the "you have been KICKED for NO RAYZION" thing as well.

**Comment by Niknakflak** [comment]
Make it raison and we gotta deal!
(Because I think the bam hammer says raison too?) Consistency.

**Comment by optimumtact** [comment]
ignore proc limits, they don't exist

File: code/game/objects/effects/mines.dm [hunk 1]
```diff
@@ -15,74 +15,52 @@
...
-	qdel(src)
 
-/obj/effect/mine/proc/explode(obj)
-	explosion(loc, 0, 1, 2, 3)
+	switch(triggertype)
+		if("explosive")
+			explosion(loc, 0, 1, 2, 3)
+		if("triggerkick")
+			if(isliving(victim) && victim.client)
+				victim << "<font color='red'><b>You have been kicked FOR NO REISIN!<b></font>"
```
Comment (Iamgoofball) [inline code/game/objects/effects/mines.dm]: Would be better if it 1:1 imperaonates the actual kick message imo

**Comment by Delimusca** [comment]
feels > reals
also good coding standards or something

File: code/game/objects/effects/mines.dm (hunk 1 above)
Comment (Delimusca) [inline code/game/objects/effects/mines.dm]: since these actually disconnect you, i think that'd be a bit much.

Comment (Iamgoofball) [inline code/game/objects/effects/mines.dm]: Nothing is ever too far.

**Comment by Jalleo** [comment]
I suggest you look at admin topic.dm its one big else if chain.
Back on topic looks nice

**Pull request merged**